ext/pg_query/extconf.rb

```diff
@@ -43,7 +43,7 @@ def ext_symbols_filename
 SYMFILE = File.join(__dir__, ext_symbols_filename)
 
 if RUBY_PLATFORM =~ /darwin/
-  $DLDFLAGS << " -Wl,-exported_symbols_list #{SYMFILE}" unless defined?(::Rubinius)
+  $DLDFLAGS << " -Wl,-exported_symbols_list #{SYMFILE}" unless defined?(::Rubinius) || RUBY_ENGINE == 'truffleruby'
```

Comment:
I wonder if we should group this together with the Rubinius check, like this:

```suggestion
  $DLDFLAGS << " -Wl,-exported_symbols_list #{SYMFILE}" unless %w(rbx truffleruby).include?(RUBY_ENGINE)
```

(didn't test it, but if we can trust the names in https://stackoverflow.com/questions/9894168/what-values-for-ruby-engine-correspond-to-which-ruby-implementations that should be correct)

Reply:
That seems to be the case ([code](https://github.com/rubinius/rubinius/blob/v5.0/core/ruby_constants.rb#L4)), but it won't build on my Mac to test.